tsrc/cli/push.py

```diff
@@ -59,23 +60,48 @@ def read_working_path(self, working_path=None):
             self.url = out
         if not self.url:
             return
+        self.tracking_ref = tsrc.git.get_tracking_ref(self.path)
         self.project_name = project_name_from_url(self.url)
         self.service = service_from_url(self.url)
 
 
 class PushAction(metaclass=abc.ABCMeta):
     def __init__(self, repository_info, args):
         self.args = args
-        self.repo_path = None
-        self.source_branch = None
-        self.target_branch = None
-        self._read_repository_info(repository_info)
+        self.repository_info = repository_info
 
-    def _read_repository_info(self, repository_info):
-        self.repo_path = repository_info.path
-        self.project_name = repository_info.project_name
-        self.source_branch = repository_info.current_branch
-        self.target_branch = self.args.target_branch
+    @property
+    def repo_path(self):
+        return self.repository_info.path
+
+    @property
+    def source_branch(self):
```

Comment:
I think 'source_branch' is misleading. We already have `current_branch`, `remote_branch` and `target_branch`